Description: fix segfault when EOF was given as password
Origin: vendor, https://bugs.gentoo.org/attachment.cgi?id=188789
Bug-Gentoo: https://bugs.gentoo.org/show_bug.cgi?id=266603
Bug-OpenSuSE: https://bugzilla.novell.com/show_bug.cgi?id=741541

## communicated by  Peter Marschall

--- a/pam_get_pass.c
+++ b/pam_get_pass.c
@@ -61,6 +61,8 @@
 	retval = conv->conv(1, msgs, &resp, conv->appdata_ptr);
 	if (retval != PAM_SUCCESS)
 		return retval;
+	if (resp[0].resp == NULL)
+		return PAM_AUTHTOK_RECOVERY_ERR;
 	retval = pam_set_item(pamh, PAM_AUTHTOK, resp[0].resp);
 	if (retval != PAM_SUCCESS)
 		return retval;
